src/main/java/com/uid2/operator/service/UIDOperatorService.java

```
@@ -68,7 +68,11 @@ public UIDOperatorService(JsonObject config, IOptOutStore optOutStore, ISaltProv
             throw new IllegalStateException(REFRESH_TOKEN_EXPIRES_AFTER_SECONDS + " must be >= " + REFRESH_IDENTITY_TOKEN_AFTER_SECONDS);
         }
 
-        this.advertisingTokenVersion = config.getBoolean("advertising_token_v3", false) ? TokenVersion.V3 : TokenVersion.V2;
+        if (config.getBoolean("advertising_token_v4", false)) {
+            this.advertisingTokenVersion = TokenVersion.V4;
+        } else {
+            this.advertisingTokenVersion = config.getBoolean("advertising_token_v3", false) ? TokenVersion.V3 : TokenVersion.V2;
```

should we throw an exception if both advertising_token_v3/advertising_token_v4 are set to true?

---

I think I prefer "newest token version wins", as that's still fairly intuitive behavior